crates/cheatnet/src/state.rs

```diff
@@ -233,7 +247,31 @@ impl CheatnetState {
 
     #[must_use]
     pub fn address_is_warped(&self, contract_address: &ContractAddress) -> bool {
-        self.warped_contracts.contains_key(contract_address)
+        self.global_warp.is_some()
+            || matches!(
+                self.warped_contracts.get(contract_address),
+                Some(WarpedContract::Warped(_))
+            )
+    }
+
+    #[must_use]
+    pub fn get_address_block_timestamp(&self, address: &ContractAddress) -> Option<Felt252> {
+        /* The Logic
+         * - If there is a global warp and no contract warp, use the global warp
+         * - If there is a global warp and a contract warp (either Warped or Unwarped), use the contract warp
+         * - If there is no global warp and a contract warp, use the contract warp (either Warped or Unwarped)
+         * - If there is no global warp and no contract warp, return None (no warp)
+         *
+         * This logic ensures that `start_warp` and `stop_warp` will always behave as expected,
+         */
```

Comment:
It can be simplified to `a specific contract being cheated takes precedence over the global warp`

Reply:
I think this kind of comment might be a bit misleading because, while it is true in this context, in the greater context of how `snforge_std::start_warp` works, calling `start_warp::(CheatTarget::All,...)` overrides individual warps by clearing the hash map.